src/codegen/struct-or-class.lisp

```diff
@@ -45,41 +45,50 @@
-                   (list))
-            ,(loop :for field :in fields
-                   :for name := (struct-or-class-field-name field)
-                   :for package := (symbol-package classname)
-                   :for accessor
-                     := (alexandria:format-symbol package "~A-~A" classname name)
-                   :collect `(,name
-                              :initform (error "")
-                              :initarg ,name
-                              :accessor ,accessor)))
+        (append 
+         (list
+          `(defclass ,classname
+               ,(if superclass
+                    (list superclass)
+                    (list))
+             ,(loop :for field :in fields
+                    :for name := (struct-or-class-field-name field)
+                    :for package := (symbol-package classname)
+                    :for accessor
+                      := (alexandria:format-symbol package "~A-~A" classname name)
+                    :collect `(,name
+                               :initform (error "")
+                               :initarg ,name
+                               :accessor ,accessor))))
+
+         (when fields
+           (list
+            `(defun ,constructor ,field-names
+               (make-instance ',classname ,@(mapcan
+                                             (lambda (field)
+                                               `(',field ,field))
+                                             field-names))))))))
+     (if (not (null fields))
+         (cons
+          `(global-lexical:define-global-lexical ,constructor
+               ,(construct-function-entry `#',constructor (length fields)))
+          (loop :for field :in fields
+                :for package := (symbol-package classname)
+                :for field-name := (alexandria:format-symbol package "~A-~A" classname (struct-or-class-field-name field))
+                :collect `(global-lexical:define-global-lexical ,field-name
+                              ,(construct-function-entry `#',field-name 1))))
 
-         `(defun ,constructor ,field-names
-            (make-instance ',classname ,@(mapcan
-                                          (lambda (field)
-                                            `(',field ,field))
-                                          field-names))))))
-     (when (not (null fields))
-       (cons
-        `(global-lexical:define-global-lexical ,constructor
-             ,(construct-function-entry `#',constructor (length fields)))
-        (loop :for field :in fields
-              :for package := (symbol-package classname)
-              :for field-name := (alexandria:format-symbol package "~A-~A" classname (struct-or-class-field-name field))
-              :collect `(global-lexical:define-global-lexical ,field-name
-                          ,(construct-function-entry `#',field-name 1))))))))
+         (list
+          `(global-lexical:define-global-lexical ,constructor
```

Comment:
`make-instance` is not defined for `structure-class` metaclasses: http://www.lispworks.com/documentation/lw70/CLHS/Body/f_mk_ins.htm

Reply:
thanks sbcl :/